plugins/chat/app/views/user_notifications/chat_summary.html.erb

```diff
@@ -2,81 +2,75 @@
   <table class="chat-summary-header text-header with-dir" style="background-color:#<%= @header_bgcolor -%>;width:100%;min-width:100%;">
     <tr>
       <td align="center" style="text-align: center;padding: 20px 0; font-family:Arial,sans-serif;">
-
         <a href="<%= Discourse.base_url %>" style="color:#<%= @header_color -%>;font-size:22px;text-decoration:none;">
           <%- if logo_url.blank? %>
             <%= SiteSetting.title %>
           <%- else %>
             <img src="<%= logo_url %>" height="40" style="clear:both;display:block;height:40px;margin:auto;max-width:100%;outline:0;text-decoration:none;" alt="<%= SiteSetting.title %>">
           <%- end %>
         </a>
-
       </td>
     </tr>
     <tr>
       <td align="center" style="font-weight:bold;font-size:22px;color:#<%= @header_color -%>">
-        <%= I18n.t("user_notifications.chat_summary.description", count: @messages.size) %>
+        <%= I18n.t("user_notifications.chat_summary.description", count: @count) %>
       </td>
     </tr>
   </table>
 
-  <%- @grouped_messages.each do |chat_channel, messages| %>
-    <%- other_messages_count = messages.size - 2 %>
-    <table class="chat-summary-content" style="padding:1em;margin-top:20px;width:100%;min-width:100%;background-color:#f7f7f7;">
-      <tbody>
-        <tr>
-          <td colspan="100%">
-            <h5 style="margin:0.5em 0 0.5em 0;font-size:0.9em;">
-              <%- if SiteSetting.private_email %>
-                <%= I18n.t("system_messages.private_channel_title", id: chat_channel.id) %>
-              <%- else %>
-                <%= chat_channel.title(@user) %>
-              <%- end %>
-            </h5>
-          </td>
-        </tr>
-
-        <%- unless SiteSetting.private_email %>
-          <%- messages.take(2).each do |chat_message| %>
-            <%- sender = chat_message.user %>
-            <%- sender_name = @display_usernames ? sender.username : sender.name %>
+  <%- [@grouped_channels, @grouped_dms].each do |grouped_messages| %>
```

Comment:
We don't have one list of all the `@grouped_messages` anymore, so iterate over both `@group_channels` and `@grouped_dms` to do the same thing.

It might change the order in which the messages are shown. Before they were globally sorted by the `created_at` of the first unread chat message. Now we prioritize all the `channels` and then all the `direct messages`.

It's also consistent with how with prioritize `channels` over `direct messages` in the subject.